src/transforms/sample/transform.rs

```diff
@@ -12,15 +12,44 @@ use crate::{
 use vector_lib::lookup::lookup_v2::OptionalValuePath;
 use vector_lib::lookup::OwnedTargetPath;
 
+/// Exists only for backwards compatability purposes so that the value of sample_rate_key is
+/// consistent after the internal implementation of the Sample class was modified to work in terms
+/// of percentages
+#[derive(Clone, Copy, Debug)]
+pub enum SampleRate {
+    OneOverN(u64),
+    Percentage(f32),
+}
```

Comment:
Is this enum even needed? The only function of it is to expose the emission ratio (via `fn pct` below), which could be pre-calculated when building `struct Sample` in the config. I guess the only issue would be the `Display` output being different for backwards compatibility.

Reply:
Yes its only used for the `Display` output, I experimented with a few options, I didn't like passing around `bool` values to `new` or more then one value for rate/ratio that may not be used. This seemed like the cleanest to me where the type constructors explicitly state the intended mode use.